### PR TITLE
ESM & Vitest

### LINKED ISSUE #197: Replace the testing environment with Vitest
## 📝 Provide a description of the improvement

In the past, we migrated several repositories from `mocha` + `mockery` stack to Vitest.

### Definition of Done

- Use `Vitest` as the testing environment tool.
- Ensure that measuring the code coverage works properly.
- Ensure CI executed tests.

### PR DESCRIPTION
**Pull request by przemyslaw-zan**
### 🚀 Summary

This PR migrates the repository to ESM, and test suite to Vitest.

---

### 📌 Related issues

* Closes #197.

---

### 💡 Additional information

*Optional: Notes on decisions, edge cases, or anything helpful for reviewers.*

### CODE REVIEW

File: tests/commands/close.js [hunk 1]
```diff
@@ -84,105 +60,112 @@ describe( 'commands/close', () => {
...
-			stubs.execCommand.execute.onCall( 0 ).resolves( {
-				logs: {
-					info: [
-						'* develop'
-					],
-					error: []
-				}
-			} );
+			let execCall = 0;
+			execCommand.execute.mockImplementation( () => {
```
Comment (psmyrek) [inline tests/commands/close.js]: Let's use mockResolvedValueOnce() (like in tests/commands/checkout.js) in all files, where we define custom output per function call.

File: tests/commands/commit.js [hunk 2]
```diff
@@ -353,29 +364,29 @@ describe( 'commands/commit', () => {
 			commandData.arguments.push( '--message' );
 			commandData.arguments.push( 'Test.' );
 
-			stubs.execCommand.execute.onFirstCall().resolves( {
+			execCommand.execute.mockResolvedValue( {
```
Comment (psmyrek) [inline tests/commands/commit.js]:
```suggestion
			execCommand.execute.mockResolvedValueOnce( {
```

File: tests/commands/push.js [hunk 3]
```diff
@@ -49,73 +28,73 @@ describe( 'commands/push', () => {
...
 		} );
 
 		it( 'skips a package if its in detached head mode', () => {
-			stubs.fs.existsSync.returns( true );
-
-			const exec = stubs.execCommand.execute;
+			fs.existsSync.mockReturnValue( true );
 
-			exec.onCall( 0 ).returns( Promise.resolve( {
+			execCommand.execute.mockReturnValue( Promise.resolve( {
```
Comment (psmyrek) [inline tests/commands/push.js]:
```suggestion
			execCommand.execute.mockReturnValueOnce( Promise.resolve( {
```

**Review by psmyrek — COMMENTED** [review]
Thanks for the changes, they look good 👍 

Unfortunately, dynamic imports `await import( <absolute path> )` does not work in Windows terminal.

Example: `node index.js status` reports: _"Command "status" does not exist. Type: "mrgit --help" in order to see available commands."_

It is caused by dynamic import, when the provided path is absolute. In fact, the app throws the following error silently, but it is not logged:

```
Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only URLs with a scheme in: file, data, and node are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'd:'
```

To fix that, we should replace the path with a file URL:

```js
import { pathToFileURL } from 'url';

const foo = await import( pathToFileURL( absolutePath ).href );
```

It should be changed in all places where dynamic import via an absolute path is used.

**Review by psmyrek — APPROVED** [review]
I checked a few commands and all look good. LGTM 🎉 

Steps I checked:

1. I created `mrgit.json` file:

    ```json
    {
        "dependencies": {
            "umberto": "cksource/umberto"
        }
    }
    ```

3. Executed `sync` command:

    ```
    node index.js sync
    
     #  umberto                                                                1/1 (100%)
    Package "umberto" was not found. Cloning...
    Cloning into 'D:/Projects/cksource/mrgit/packages/umberto'...
    remote: Enumerating objects: 34806, done.
    remote: Counting objects: 100% (195/195), done.
    remote: Compressing objects: 100% (147/147), done.
    remote: Total 34806 (delta 83), reused 89 (delta 36), pack-reused 34611 (from 2)
    Receiving objects: 100% (34806/34806), 15.79 MiB | 2.26 MiB/s, done.
    Resolving deltas: 100% (20694/20694), done.
    
    1 packages have been processed.
    Execution time: 10s376ms.
    ```

4. Executed `status` command.

    ```
    $ node index.js status
    
    Collecting statuses...
    ┌─────────┬────────────┬─────────┬────────┐
    │ Package │ Branch/Tag │ Commit  │ Status │
    ├─────────┼────────────┼─────────┼────────┤
    │ umberto │ master     │ b500a49 │        │
    └─────────┴────────────┴─────────┴────────┘
    Legend:
    ↑ branch is ahead ↓ or behind, + staged files, M modified files, ? untracked files,
    L latest tag, ! not on a branch or a tag specified in configuration file.
    Execution time: 0s731ms.
    ```
